packages/server/src/automations/utils.ts

```diff
@@ -14,13 +14,16 @@ import sdk from "../sdk"
 const REBOOT_CRON = "@reboot"
 const WH_STEP_ID = definitions.WEBHOOK.stepId
 const CRON_STEP_ID = definitions.CRON.stepId
-const Runner = new Thread(ThreadType.AUTOMATION)
 
 const jobMessage = (job: any, message: string) => {
   return `app=${job.data.event.appId} automation=${job.data.automation._id} jobId=${job.id} trigger=${job.data.automation.definition.trigger.event} : ${message}`
 }
 
-export async function processEvent(job: any) {
+export async function processEvent(job: any, timeout?: { timeout?: number }) {
+  const Runner = new Thread(ThreadType.AUTOMATION, {
```

Comment:
@mike12345567 I've moved the thread initialisation inside this function as it now needs access to the timeout which is coming through, does this make sense?

Reply:
hmm I think this could be pretty dangerous - this technically means its creating a thread per event - we are trying to avoid that with total initialisation.

I think a better approach would be to avoid the whole thread mechanism, and then just use a promise timeout (`Promise.race`) - so essentially load the `execute` function directly for sync runs instead - if you need more info/thoughts are this give me a shout and we can have a call!

Reply:
Added a flag inside the orchestrator so we exit successfully.